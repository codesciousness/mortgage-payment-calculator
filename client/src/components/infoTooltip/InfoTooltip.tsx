import React, { useState } from 'react';
import Tooltip from '@mui/material/Tooltip';
import InfoIcon from '@mui/icons-material/Info';
import IconButton from '@mui/material/IconButton';
import ClickAwayListener from '@mui/material/ClickAwayListener';

type InfoTooltipProps = {
    title: string;
};

const InfoTooltip = ({ title }: InfoTooltipProps): JSX.Element => {
    const [open, setOpen] = useState(false);

    let id = title.split(' ', 2).join('');
    id = id.slice(0, id.length - 1);

    const handleTooltipClose = () => {
        setOpen(false);
    };
    
    const handleTooltipOpen = () => {
        setOpen(true);
    };

    return (
        <ClickAwayListener onClickAway={handleTooltipClose}>
            <Tooltip
                id={id}
                title={title}
                open={open}
                placement='top-end'
                arrow
                disableFocusListener
                disableHoverListener
                disableTouchListener
                sx={{
                    color: 'DeepSkyBlue',
                    float: 'right',
                    padding: 0
                }}
            >
                <IconButton onClick={handleTooltipOpen}>
                    <InfoIcon/>
                </IconButton>
            </Tooltip>
        </ClickAwayListener>
    );
};

export default InfoTooltip;