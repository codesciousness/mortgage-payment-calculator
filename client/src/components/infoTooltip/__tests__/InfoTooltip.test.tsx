import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import InfoTooltip from '../InfoTooltip';

const title = 'Title';

describe('<InfoTooltip />', () => {
    test('InfoTooltip component is defined', () => {
        expect(<InfoTooltip title={title}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<InfoTooltip title={title}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly when title prop is provided', () => {
        render(<InfoTooltip title={title}/>);
        expect(screen.queryByText('Title')).not.toBeInTheDocument();
    });
    test('displays title when button is clicked', () => {
        render(<InfoTooltip title={title}/>);
        fireEvent.click(screen.getByRole('button'));
        expect(screen.getByText('Title')).toBeInTheDocument();
    });
});