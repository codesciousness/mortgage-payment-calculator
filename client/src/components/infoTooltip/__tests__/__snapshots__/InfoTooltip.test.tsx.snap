// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoTooltip /> component matches the snapshot 1`] = `
<button
  aria-label="Title"
  aria-labelledby={null}
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-123drkj-MuiButtonBase-root-MuiIconButton-root"
  data-mui-internal-clone-element={true}
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="InfoIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
